apps/web/src/components/QualifiedRecruitmentCard/QualifiedRecruitmentCard.tsx

```diff
@@ -51,7 +50,7 @@ const QualifiedRecruitmentCard = ({
 
   // NOTE: Until we store assessed skills, we will just be displayed all essential skills
   const categorizedSkills = categorizeSkill(
-    candidate.pool.essentialSkills?.filter(notEmpty) ?? [],
+    poolSkillsToSkills(candidate.pool.poolSkills),
```

Comment:
It was essentialSkills before, now its all poolSkills, is that okay?

Reply:
I think we should keep it as before and narrow it to essential skills

Reply:
So, this is where it gets tricky :sweat_smile: It technically is still essential because that is what we are querying. 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e942c8e4a8caf92ded9d13c263eeceb403ac23a7/apps/web/src/pages/Profile/CareerTimelineAndRecruitmentPage/components/CareerTimelineAndRecruitment.tsx#L136

However, what I wanted to do was do a fragment mask for this component so we don't need to rely on the query.

Though, that makes things weird since this is a pool candidate thing that is getting masked in the other PR. But, I get where you are both coming from, it could be far more explict. I'm struggling where to do this work though, here or  in the other PR :thinking:

Reply:
I think my suggestion for now would be to run this through `filterPoolSkillsByType`, even if we know it should only be essential from the query. That makes the code a bit clearer, and guards against changes to the query. And leave a comment explaining we're being deliberately redundant, so future us doesn't think it's just an oversight.

Reply:
okay, updated in [update to be explicit about essential skills](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10224/commits/c65ff37f5357b484f6e8627b2719e06432210fba)